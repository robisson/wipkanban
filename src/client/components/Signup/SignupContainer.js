import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onCreateAccount(email, password) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = ({
  userReducer: { success, message, showPreloader }
}) => {
  return {
    success,
    message,
    showPreloader
  };
};

const SignupContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup);

export default SignupContainer;
