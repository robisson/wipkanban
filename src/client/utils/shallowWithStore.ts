import { shallow } from "enzyme";

export default (component:any, store:any) => {
  const context = {
    store
  };
  return shallow(component, { context });
};
