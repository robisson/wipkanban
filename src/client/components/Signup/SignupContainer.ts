import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";
import { Dispatch } from "redux";

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onCreateAccount(email: string, password: string) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = ({
  userReducer: { success, message, showPreloader }
}: any) => {
  return {
    success,
    message,
    showPreloader
  };
};

const SignupContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Signup);

export default SignupContainer;
