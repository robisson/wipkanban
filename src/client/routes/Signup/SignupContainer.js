import { connect } from "react-redux";
import { createAccount } from "../../actions/User";
import Signup from "./index";

const mapDispatchToProps = dispatch => {
  return {
    onCreateAccount(email, password) {
      dispatch(createAccount(email, password));
    }
  };
};

const mapStateToProps = state => {
  return { state: state.userReducer };
};

const SignupContainer = connect(mapStateToProps, mapDispatchToProps)(
  Signup
);

export default SignupContainer;
