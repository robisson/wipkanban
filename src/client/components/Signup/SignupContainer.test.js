import React from "react";
import toJson from "enzyme-to-json";
import "../../setupTest";
import SignupContainer from "./SignupContainer";
import shallowWithStore from "../../utils/shallowWithStore";
import { createMockStore } from "redux-test-utils";

describe("<SignupContainer /> Component", () => {
  let wrapper;

  it("should to render correctly", () => {
    const testState = {
      userReducer: {
        message: "",
        showPreloader: false,
        success: false,
        onCreateAccount: jest.fn()
      }
    };

    const store = createMockStore(testState);
    wrapper = shallowWithStore(<SignupContainer />, store);
    expect(wrapper.dive().props()).toHaveProperty("message");
    expect(wrapper.dive().props()).toHaveProperty("showPreloader");
    expect(wrapper.dive().props()).toHaveProperty("success");
    expect(wrapper.dive().props()).toHaveProperty("onCreateAccount");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
