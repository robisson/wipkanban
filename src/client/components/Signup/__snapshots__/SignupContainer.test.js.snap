// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupContainer /> Component should to render correctly 1`] = `
<WithStyles(Signup)
  message=""
  onCreateAccount={[Function]}
  showPreloader={false}
  success={false}
/>
`;
